src/CFNetwork/CFHTTPMessage.cs

```diff
@@ -85,6 +85,11 @@ static IntPtr GetVersion (Version version)
 					return _HTTPVersion2_0.Handle;
 				// HTTP 2.0 requires OS X 10.11 or later.
 				return _HTTPVersion1_1.Handle;
+			} else if (version.Major == 3 && version.Minor == 0) {
+				if (_HTTPVersion3_0 != null && _HTTPVersion3_0.Handle != IntPtr.Zero)
+					return _HTTPVersion3_0.Handle;
+				// HTTP 3.0 requires OS X 10.16 or later.
+				return _HTTPVersion1_1.Handle;
```

Comment:
Shouldn't the fallback be HTTP/2 and not HTTP/1.1 here (and then it should check if HTTP/2 is supported first, like the code block just above)?

Reply:
You are right.. will change the entire if to got from max to min

Reply:
yup, looks like you should check for it earlier
so if 3.0 is not present, then 2.0 is checked and if not present 1.1 can be used (e.g. on 10.10)